docker_cli/_docker_cuda.py

```diff
@@ -0,0 +1,284 @@
+
+    def _generate_initial_lines(self, repo_ver: str, *, arch: str = "x86_64") -> str:
+        cuda_repo_name = (
+            "https://developer.download.nvidia.com/"
+            f"compute/cuda/repos/{repo_ver}/{arch}/"
+        )
+        filename = f"cuda-keyring_1.0-1_all.{self.package_manager.file_type}"
+        read_section = self.url_reader.generate_read_command(
+            target=f"{cuda_repo_name}" + filename,
+            output_file=filename,
+        )
+        package_command_section = self.package_manager.generate_package_command(
+            target=filename,
+        )
+        return f"RUN {read_section} \\\n && {package_command_section}"
+
+    def generate_install_lines(self, build_targets: Iterable[str]) -> str:
+        targets = build_targets
+        install_section = self.package_manager.generate_install_command(
+            targets=targets, clean=True
+        )
+        install_line = (
+            f"RUN {self.package_manager.name} update \\\n && {install_section}"
+        )
+        return install_line
+
+
+class YumCUDADockerfileGen(CUDADockerfileGenerator):
+    """
+    A CUDA dockerfile generator for rpm-based yum systems
+    """
+
+    _dev_build_targets = [
+        "cuda-cudart-devel-$CUDA_PKG_VERSION",
+        "cuda-nvcc-$CUDA_PKG_VERSION",
+        "libcufft-devel-$CUDA_PKG_VERSION",
+        "rpm-build",
+    ]
+
+    def __init__(self, url_reader: URLReader):
+        super(YumCUDADockerfileGen, self).__init__(url_reader=url_reader)
+        self._package_manager: PackageManager = get_package_manager("yum")
+
+    def _generate_initial_lines(self, repo_ver: str, *, arch: str = "x86_64") -> str:
+        cuda_repo_name = (
+            "https://developer.download.nvidia.com/"
+            f"compute/cuda/repos/{repo_ver}/{arch}/"
+        )
+        cuda_repo = f"{cuda_repo_name}cuda-{repo_ver}.repo"
+        return f"RUN yum-config-manager --add-repo {cuda_repo} \\\n " "&& yum clean all"
+
+    def generate_install_lines(self, build_targets: Iterable[str]) -> str:
+        targets = build_targets
+        install_line = self.package_manager.generate_install_command(
+            targets=targets, clean=True
+        )
+        install_line = "RUN " + install_line
+        return install_line
+
+
+def get_cuda_dockerfile_generator(
+    pkg_mgr: PackageManager | str, url_reader: URLReader
+) -> CUDADockerfileGenerator:
+    """
+    Returns the appropriate dockerfile generator for a system.
+
+    Parameters
+    ----------
+    pkg_mgr : PackageManager or str
+        The package manager in use by the image
+    url_reader : URLReader
+        The URL reader (e.g. cURL, wget) in use by the image
+    origin_image_name : str
+        The name of the image on which this one is based (e.g. "ubuntu") -
+        note that this is the very basis image, not the immediate parent image.
+    arch : str, optional
+        The architecture of the CUDA repository, by default "x86_64"
```

Comment:
Doesn't look like `origin_image_name` or `arch` are parameters accepted by this function right now.

```suggestion
```